src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/FilterColumnsWithAllNullValues.java

```diff
@@ -28,7 +30,7 @@ public static void main(String[] args) {
             .setOutput(pipelineOutput);
 
         Dataset dataset = new MemoryDataset(pipeline);
-        dataset.load().waitForRecordsToLoad();
+        dataset.load().waitForRecordsToLoad(100, TimeUnit.SECONDS.toMillis(4));
```

Comment:
Can you explain why `waitForRecordsToLoad()` doesn't work, but `waitForRecordsToLoad(100, TimeUnit.SECONDS.toMillis(4))` does?  The original call should wait until everything is loaded. 🤔

Reply:
I changed back this line as you and Gautam said. Added an additional code to wait for column stats to load.